manifest: Update Softdevice controller & MPSL (#4253)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

.. rst-class:: v1-5-0

> DRGN-15465: Corrupted advertising data when :option:`CONFIG_BT_EXT_ADV` is set
>   Setting scan response data using legacy HCI commands on a build with extended advertising support corrupts parts of the advertising data.
>   Zephyr sets the complete local name in the scan response data by default, and the corrupted advertising data may confuse some scanners causing the device to not be visible.

Reviewer: ```suggestion
DRGN-15465: Corrupted advertising data when extended advertising support is enabled
  Setting scan response data using legacy HCI commands when :option:`CONFIG_BT_EXT_ADV` is set corrupts parts of the advertising data.
  Zephyr sets the complete local name in the scan response data by default.
  The corrupted advertising data might confuse some scanners, which causes the device to not be visible.
```

Reviewer: Not sure I understand where the complete local name comes in here? Can you clarify this?

Reviewer 2: I think the second sentence could be removed, it explains some of the symptoms, but the sentence above describes the whole issue.

Or it could be rephrased to say that when using BT_LE_ADV_OPT_USE_NAME the name is put in the scan response which would then be corrupted.

Complete local name is an AD (Advertising Data) type.
However the type could also be Shortened local name. So best to just say "device name" instead, which is the term used in the API.

Author: Yes, I'll reword it. I think it's useful to mention that this is linked to advertising the name of the device, so the users know that it is this issue that is affecting them.

Reviewer: Yes, it makes sense to have that information, but there should be a clear connection - right now it doesn't say how this causes a problem.

Author: I just updated the message, lemme know if it makes sense to you

>   **Workaround:** Upgrade to v1.5.1 (once available) or use the master branch.

Reviewer 2: Workarounds should be a way to fix the problem on the version that is affected.
If there is no known workaround then don't include one.

The one below shouldn't have been worded like this either.

Reviewer: Agree that this is not a suitable workaround here (but it's okay for the issue below, since that is an issue for one version only and the only workaround is "do not use this version").

For this issue: Can you set the scan response data in some other way? Or configure Zephyr differently (since the local name is set "by default")?

>   Setting scan response data using legacy HCI commands on a build with extended advertising support corrupts parts of the advertising data.
>   When using BT_LE_ADV_OPT_USE_NAME (default configuration in most samples), the name is put in the scan response, which would then corrupt the advertising data.
> 
>   **Workaround:** Don't set scan response data. That implies not using the BT_LE_ADV_OPT_USE_NAME option, and instead manually setting the device name in the advertising data if necessary. e.g. do not use the default BT_LE_ADV_CONN_NAME configuration when initializing bluetooth, but rather use BT_LE_ADV_CONN.

Reviewer: ```suggestion
  Setting scan response data using legacy HCI commands on a build with extended advertising support corrupts parts of the advertising data.
  When using ``BT_LE_ADV_OPT_USE_NAME`` (which is the default configuration in most samples), the device name is put in the scan response.
  This corrupts the advertising data.

  **Workaround:** Do not set scan response data.
  That implies not using the ``BT_LE_ADV_OPT_USE_NAME`` option or the :c:macro:`BT_LE_ADV_CONN_NAME` macro when initializing Bluetooth.
  Instead, set the device name manually in the advertising data if necessary by using :c:macro:`BT_LE_ADV_CONN`.
```
Please check the workaround again, not sure if I got it right.

.. rst-class:: v1-5-0

DRGN-15465: Corrupted advertising data when :option:`CONFIG_BT_EXT_ADV` is set
  Setting scan response data using legacy HCI commands on a build with extended advertising support corrupts parts of the advertising data.
  When using ``BT_LE_ADV_OPT_USE_NAME`` (which is the default configuration in most samples), the device name is put in the scan response.

>   This corrupt the advertising data.

Reviewer: ```suggestion
  This corrupts the advertising data.
```

.. rst-class:: v1-5-0

DRGN-15465: Corrupted advertising data when :option:`CONFIG_BT_EXT_ADV` is set

>   Setting scan response data using legacy HCI commands on a build with extended advertising support corrupts parts of the advertising data.

Reviewer 2: @rugeGerritsen @jori-nordic  This doesn't match my understanding of the issue.
Isn't it it setting scan response data for a legacy advertiser using extended HCI commands?

Author: Would this be better then:
Setting scan response data for a legacy advertiser on a build with extended advertising support corrupts parts of the advertising data.

Reviewer 2: Yes, that seems correct.

Reviewer 3: Shouldn't it rather be:
Setting scan response data for a legacy advertiser using extended advertising commands corrupts parts of the advertising data.
?
When setting scan response data on a build with extended advertising support using legacy commands works just fine.

Author: Yes but this part right there in hci_driver.c will ensure we only use the extended advertising commands from the zephyr host right ?
```c
	if (IS_ENABLED(CONFIG_BT_BROADCASTER)) {
		if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT)) {
			err = sdc_support_ext_adv();
```

Author: what Im trying to say is that for all intents and purposes, this sentence correctly describes the SDC behavior when integrated with zephyr and when enabling ext adv.

  When using ``BT_LE_ADV_OPT_USE_NAME`` (which is the default configuration in most samples), the device name is put in the scan response.
  This corrupts the advertising data.

  **Workaround:** Do not set scan response data.

>   That implies not using the ``BT_LE_ADV_OPT_USE_NAME`` option, or the :c:macro:`BT_LE_ADV_OPT_USE_NAME` macro when initializing Bluetooth.

Reviewer: ```suggestion
  That implies not using the ``BT_LE_ADV_OPT_USE_NAME`` option, or the :c:macro:`BT_LE_ADV_CONN_NAME` macro when initializing Bluetooth.
```
That's what you put originally, not sure why you changed it?

Author: hadn't eaten yet :smile:
